RELEASE/scripts/autoscend/quests/level_13.ash

```diff
@@ -1671,6 +1254,176 @@ boolean L13_towerNSTower()
 	return false;
 }
 
+boolean L13_towerNSTowerSkin()
+{
+	if(!contains_text(visit_url("place.php?whichplace=nstower"), "ns_05_monster1"))
+	{
+		return false;
+	}
+	auto_log_info("Time to fight the Wall of Skins!", "blue");
+	if (get_property("auto_towerBreak").to_lower_case() == "wall of skin" || get_property("auto_towerBreak").to_lower_case() == "wallofskin" || get_property("auto_towerBreak").to_lower_case() == "skin" || get_property("auto_towerBreak").to_lower_case() == "level 1")
+	{
+		abort("auto_towerBreak set to abort here.");
+	}
+	if (item_amount($item[Beehive]) > 0)
+	{
+		return autoAdvBypass("place.php?whichplace=nstower&action=ns_05_monster1", $location[Tower Level 1]);
+	}
+	// Can we kill the tower without a beehive?
+	beehiveConsider(true);
+	if(get_property("auto_getBeehive").to_boolean())
+	{
+		return false;
+	}
+	
+	int damage = 2; // base attack damage plus TT attack skill (kneebutt, headbutt)
+	
+	boolean fam_set = false;
+	foreach fam in $familiars[shorter-order cook, mu, imitation crab] // crab is evergreen, buy one
+	{
+		if (have_familiar(fam) && auto_is_valid(fam))
+		{
+			handleFamiliar(fam);
+			use_familiar(fam);
+			damage += 4; // worst one is crab, at 4.
+			fam_set = true;
+			break;
+		}
+	}
+	if (!fam_set) // just use some trash that does damage that we will have
+	{
+		foreach fam in $familiars[angry goat, MagiMechTech MicroMechaMech, star starfish, mosquito]
+		{
+			if (have_familiar(fam) && auto_is_valid(fam))
+			{
+				handleFamiliar(fam);
+				use_familiar(fam);
+				break;
+			}
+		}
+	}
+		
+	// We've probably got a tiny bowler, that'll help.
+	if (available_amount($item[tiny bowler]) > 0 && can_equip($item[tiny bowler]))
+	{
+		autoEquip($item[tiny bowler]);
+		damage += 1; //familiar attack
+	}
+	
+	// apply skills
+	// start by shrugging unnecessary AT skills
+	// These ones should be safe to just remove simply
+	uneffect($effect[Aloysius' Antiphon of Aptitude]);
+	uneffect($effect[Ode to Booze]);
+	uneffect($effect[The Sonata of Sneakiness]);
+	uneffect($effect[Carlweather\'s Cantata of Confrontation]);
+	uneffect($effect[Cletus\'s Canticle of Celerity]);
+	
+	// TODO: These need to be handled so they're not recast
+	uneffect($effect[Ur-Kel\'s Aria of Annoyance]);
```

Comment:
want to uneffect Blood Bubble too so first hit isn't blocked

Reply:
Added without testing - I guess this will need an SGEEA? Although these are cheap/plentiful now olfaction doesn't need them.

Reply:
yeah. This was handled by the original code: if you miss the first enemy hit you don't get first round thorns damage